components/filter-components/TimelineRangeSelector.vue

```diff
@@ -117,27 +113,32 @@ export default {
     }
   },
   watch: {
-    recommendedRange(newRecommendedRange, oldQuestion) {
+    recommendedRange(newRecommendedRange) {
       if (newRecommendedRange !== null) {
-        let index = newRecommendedRange.index;
+        let index = newRecommendedRange;
         let dateRange = this.panicsRanges[index];
         let startRange = [dateRange[0], dateRange[1]];
         this.$emit("scrollHeaderIntoView")
         this.$emit("selectedDateRange", [dateRange, startRange]);
       }
     },
     customInputRange(newCustomInputRange) {
-      if (newCustomInputRange) {
-        this.$emit("selectedDateRange", [newCustomInputRange, newCustomInputRange]);
-      } else {
-        this.$emit("selectedDateRange", [this.defaultRangeCustom, this.defaultRangeCustom])
+      if (newCustomInputRange && newCustomInputRange.length === 2) {
```

Comment:
Should we be doing anything if these validation checks fail? Can they fail?

Reply:
if it fails it just shouldn't emit anything.
customInputRange(newCustomInputRange) {
      if (newCustomInputRange && newCustomInputRange.length === 2) {
        //Converting strings to date format
        const parsedDates = newCustomInputRange.map(dateStr => {
          return new Date(dateStr); 
        });
        this.$emit("selectedDateRange", [parsedDates,parsedDates]);
      }
      return;
    }
    
    is this better with an empty return?
    
    it fails when first using the the custom time line selector with the date picker.

Reply:
Chatted with Ruby and she's confirmed the check is only there to make sure that the date range is triggered once the user has selected both a start and end date for the range. Makes sense and I don't think any changes are required